applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationClientTest.java

```diff
@@ -620,4 +938,26 @@ private static void assertConfigurationEquals(ConfigurationSetting expected, Con
             });
         }
     }
+    
+    /**
+     * Helper method to verify the error was a RestException and it has a specific HTTP response code.
+     * @param ex Expected error thrown during the test
+     * @param expectedStatusCode Expected HTTP status code contained in the error response
+     */
+    private static void assertRestException(Throwable ex, int expectedStatusCode) {
+        assertTrue(ex instanceof RestException);
+        assertEquals(expectedStatusCode, ((RestException) ex).response().statusCode());
+    }
+    
+    /**
+     * Helper method to verify that a command throws an IllegalArgumentException.
+     * @param exceptionThrower Command that should throw the exception
+     */
+    private static void assertRunnableThrowsArgumentException(Runnable exceptionThrower) {
+        try {
+            exceptionThrower.run();
+        } catch (Exception ex) {
```

Comment:
If the runner doesn't throw, it'll pass. I'd add an assert fail of sorts after the .run();
It might be better to catch IllegalArgumentException rather than the global Exception.

Reply:
Makes sense to fail if an error isn't thrown, I'll add that. Not sold on specifically catching IllegalArgumentException, the test should fail if it catches something else (however unlikely)

Reply:
If it doesn't catch that exception, the test will fail with an unhandled exception.